test/plugins/scenarios/fasthttp/main.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package main
+
+import (
+	"github.com/valyala/fasthttp"
+	"time"
+
+	_ "github.com/apache/skywalking-go"
+)
+
+func providerHandler(ctx *fasthttp.RequestCtx) {
+	ctx.WriteString("success")
+}
+
+func consumerHandler(ctx *fasthttp.RequestCtx) {
+	url := "http://localhost:8080/provider"
+
+	req := fasthttp.AcquireRequest()
+	req.SetRequestURI(url)
+
+	resp := fasthttp.AcquireResponse()
+
+	timeout := 5 * time.Second
+	err := fasthttp.DoTimeout(req, resp, timeout)
+
+	if err != nil {
+		ctx.Error("Internal Server Error", fasthttp.StatusInternalServerError)
+		return
+	}
+
+	ctx.SetStatusCode(resp.StatusCode())
+	ctx.Write(resp.Body())
+
+	fasthttp.ReleaseRequest(req)
+	fasthttp.ReleaseResponse(resp)
+}
+
+func main() {
+	fasthttp.ListenAndServe(":8080", func(ctx *fasthttp.RequestCtx) {
```

Comment:
Is the `fasthttp` is HTTP server or client framework? In this demo, it appears to be both a server and a client at the same time. However, your plugin code only provides interception for HTTP clients?

Reply:
Thank you for your review.
We do not use the server in our project, so we only write the enhancement to the client. 
I'll add it later.

Reply:
Got it, will you add it in this PR or a new one?

Reply:
In this PR, I will supplement the code in the next few days